components/public-api/gitpod/v1/organization.proto

```diff
@@ -151,6 +169,23 @@ message TimeoutSettings {
 }
 
 message UpdateOrganizationSettingsRequest {
+  message OnboardingSettings {
```

Comment:
unfortunately, I had to make a request / response split for the types here, so there's duplication.

This is because of the following fields:
- `featured_member_resolved_avatar_url` is only available in responses
- `update_recommended_repositories` is only available in requests

Reply:
Can-do. My motivation here was to have a properly defined proto for both, where you don't have un-settable properties as the types. I will then unify the `OnboardingSettings` shapes and add appropriate comments to explain their specifics.

Reply:
> My motivation here was to have a properly defined proto for both, where you don't have un-settable properties as the types.

Personally I'm totally on the same side: I also want to definitely declare what's there and what not. But since at least proto3, protobuf is no longer meant to be something that enforces any kind of "protocol", it's merely translation tool. That shows in that you can't guarantee a field being present anymore, and instead have to resort to dynamic checking "is field present?" in your code anyway - so we actually follow the intention by doing that. :slightly_smiling_face: 
So by treating fields individually and just checking them when necessary, we

Reply:
@geropl pushed [`0e3fba2` (#20577)](https://github.com/gitpod-io/gitpod/pull/20577/commits/0e3fba230ad766cee35f3a206b084f8657067316) to unify.